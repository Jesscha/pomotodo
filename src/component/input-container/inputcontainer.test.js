import React from 'react';
import { shallow } from 'enzyme';
import InputContainer from './Inputcontainer.component';







describe("<InputContainer/ >", ()=>{
    const wrapper = shallow(<InputContainer/>)
    beforeEach(()=>{
        wrapper.setState ({
            id: 1,
            todoInput: "abc",
            pomoCount: 4
        });
    });


    it("renders InpunContainer", ()=>{
        expect(wrapper).toMatchSnapshot();
    });

    it("each input shows each state", ()=>{
       
        const todoInput = wrapper.find('#todo-input');
        const pomoCount = wrapper.find('#pomo-count');
        expect(todoInput.prop('value')).toEqual("abc");
        expect(pomoCount.prop('value')).toEqual(4);
    })
    it("initialize state when button clicked", ()=>{
        wrapper.find('button').simulate("click")
        expect(wrapper.state().todoInput).toEqual("")
        expect(wrapper.state().pomoCount).toEqual(1)
    })

    
    })





