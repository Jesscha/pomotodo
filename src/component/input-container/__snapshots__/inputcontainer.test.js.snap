// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputContainer/ > renders InpunContainer 1`] = `ShallowWrapper {}`;
